unity/Assets/Scripts/AgentManager.cs

```diff
@@ -728,147 +704,184 @@ private void addImageSynthesisImage(List<KeyValuePair<string, byte[]>> payload,
     }
 
 
-	public IEnumerator EmitFrame() {
+    private bool canEmit() {
+        bool emit = true;
+        foreach (BaseFPSAgentController agent in this.agents) {
+            if (agent.agentState != AgentState.Emit) {
+                emit = false;
+                break;
+            }
+        }
 
-		frameCounter += 1;
+        return this.agentState == AgentState.Emit && emit;
```

Comment:
why do you check for `this.agentState == AgentState.Emit` is it the primary agent? shouldn't it be in `this.agents` ?

Reply:
this.agentState (on AgentManager) should get renamed.  Since the agentManager also transitions through states because it handles actions, its state must also be checked.